### PR TITLE
[libc][cmake] make i386 distinct from x86_64

### PR DESCRIPTION
**Pull request by nickdesaulniers**
Configured via:

    $ cmake ../runtimes -G Ninja -DLLVM_ENABLE_LLD=ON \
      -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_RUNTIMES="libc" \
      -DLIBC_TARGET_TRIPLE=i386-linux-gnu -DCMAKE_C_COMPILER=clang \
      -DCMAKE_CXX_COMPILER=clang++

Link: #93709

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
FWIW, I do plan to flush out the entrypoints and headers text files, but not in this PR.

**Review by lntue — APPROVED** [review]

**Review by michaelrj-google — APPROVED** [review]
LGTM, but we should plan to stand up a bot for this i386 once it's properly working.

**Review by petrhosek — APPROVED** [review]

File: libc/cmake/modules/LLVMLibCCheckCpuFeatures.cmake [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # Initialize ALL_CPU_FEATURES as empty list.
 set(ALL_CPU_FEATURES "")
 
-if(${LIBC_TARGET_ARCHITECTURE_IS_X86})
+if(${LIBC_TARGET_ARCHITECTURE_IS_X86_64})
```
Comment (jhuber6) [inline libc/cmake/modules/LLVMLibCCheckCpuFeatures.cmake:8]: Why do we need `${...}` here?

**Review by jhuber6 — COMMENTED** [review]
Makes sense in general, guessing we'll fill out the entrypoints later.

**Review by jhuber6 — APPROVED** [review]

**Pull request merged**